src/main/java/org/rumbledb/compiler/VisitorHelpers.java

```diff
@@ -20,16 +23,33 @@ public static void resolveDependencies(Node node, RumbleRuntimeConfiguration con
         new VariableDependenciesVisitor(conf).visit(node, null);
     }
 
-    public static void populateStaticContext(Node node) {
+    private static void printTree(Node node, RumbleRuntimeConfiguration conf) {
+        if (conf.isPrintIteratorTree()) {
+            System.out.println("***********************");
+            System.out.println("Initial expression tree");
+            System.out.println("***********************");
+            System.out.println("Unset execution modes: " + node.numberOfUnsetExecutionModes());
+            System.out.println(node);
+            System.out.println();
+        }
+    }
+
+    public static void populateStaticContext(Node node, RumbleRuntimeConfiguration conf) {
+        printTree(node, conf);
         StaticContextVisitor visitor = new StaticContextVisitor();
         visitor.visit(node, null);
 
+
         visitor.setVisitorConfig(VisitorConfig.staticContextVisitorIntermediatePassConfig);
         int prevUnsetCount = Functions.getUserDefinedFunctionIdentifiersWithUnsetExecutionModes().size();
+        printTree(node, conf);
```

Comment:
I think the printTree() function should do one thing that is to actually print the tree. I believe moving `if (conf.isPrintIteratorTree()) {` check into that function does more harm than good in terms of readability. As such, the code has the illusion that it always prints the iterator tree and only when you check the printTree() function you get the idea that it is in fact not printed thanks to the configuration.

Reply:
Good catch, done.